specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2016-09-01/servicefabric.json

```diff
@@ -356,6 +395,96 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/environments/{environment}/clusterVersions/{clusterVersion}": {
+      "get": {
+        "tags": [
+          "ClusterVersion"
+        ],
+        "operationId": "ClusterVersions_GetByEnvironment",
+        "description": "Get cluster code versions by environment",
+        "x-ms-examples": {
+          "Get cluster version": {
+            "$ref": "./examples/ClusterVersionsGetByEnvironment_example.json"
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/locationForClusterCodeVersions"
+          },
+          {
+            "$ref": "#/parameters/environment"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/clusterVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Get cluster code versions successfully",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
```

Comment:
From the resource provider guidelines, I see that a GET response on a URL ending with a resource name must return a JSON object. I'm attaching the part of the document below.

## GetResource

Returns a resource belonging to a resourcegroup. Resource types can be nested and, if so, must follow the RESTguidelines (full details in the nested resource type section). 

### Request

| Method | Request URI                              |
| ------ | ---------------------------------------- |
| GET    | https://<endpoint>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}?api-version={api-version} |

 

**Arguments**

| Argument          | Description                              |
| ----------------- | ---------------------------------------- |
| subscriptionId    | The subscriptionId for  the Azure user.  |
| resourceGroupName | The resource group  name uniquely identifies the resource group within the user subscriptionId. |
| resourceType      | The type of the  resource – the resource providers declare the resource types they support at  the time of registering with Azure. |
| resourceName      | The name  of the resource.               |
| api-version       | Specifies  the version of the protocol used to make this request. |

 

**Request Headers**

Only headers common to allrequests.

**Request Body**

Empty

### Response

The response includes an HTTP status code, a set ofresponse headers, and a response body.

 

**Status Code**

The resource provider should return 200 (OK) to indicatethat the operation completed successfully.

 

If the resource does not exist, 404 (NotFound) should bereturned. For other errors (e.g. internal errors) use the appropriate HTTPerror code.

 

**Response Headers**

Headers common to all responses.

 

**Response Body**

{

  “id”: “/subscriptions/{id}/resourceGroups/{group}/providers/{rpns}/{type}/{name}”,

  “name”: “{name}”,

  “type”: “{resourceProviderNamespace}/{resourceType}”,

  "location":"North US",

  "tags": {

​    "key1": "value 1",

​    "key2": "value 2"

   },

 

  “etag”: “00000000-0000-0000-0000-000000000000”,

  "properties":{ "comment: "Resource defined structure" }

}

 

For a detailed explanationof each field in the response body, please refer to the request bodydescription in the PUT resource section. The only GET specific properties are“name,” “type” and “id.”

| Field | Description                              |
| ----- | ---------------------------------------- |
| id    | Required,  string  The id  field should be the URL (excluding the hostname/scheme and api version) for  the entity. It should not be URL encoded. E.g.   /subscriptions/{id}/resourceGroups/{rgName}/providers/{rpns}/{typeName}/{name}     This  field is important to the platform – it is used as the identifier for references  on other objects (e.g. if a virtual machine “points” to a vNet, it uses the  id of the vNet as its reference), displaying links / references between  resources in the portal, Authorization checks / validation, auditing /  operational logs, etc. |
| name  | Required,  string  The name  does not need to be URL encoded or match exactly what is seen in the URL.     It  should be the name of the resource and is not expected to be globally unique  (only unique for that particular collection / type in the resource group). |
| type  | Required,  string  The type  does not need to be URL encoded or match exactly what is seen in the URL.      It  should include the resource provider namespace *and* the type of the entity.  Examples include Microsoft.Web/Sites. |
| etag  | Optional,  string  The Etag  field is *not* required. If it is provided in the response body, it must also  be provided as a header per [the normal ETag convention](http://www.rfc-editor.org/rfc/rfc2616.txt).      Entity tags are used for comparing  two or more entities from the same requested resource. HTTP/1.1 uses entity  tags in the ETag (section 14.19), If-Match (section 14.24), If-None-Match  (section 14.26), and If-Range (section 14.27) header fields.     The full guidance can be found [in the Addendum](#_ETags_for_Resources). |